osu.Game/Rulesets/Difficulty/DifficultyAttributes.cs

```diff
@@ -64,6 +64,16 @@ public class DifficultyAttributes
         /// </summary>
         public double LegacyBonusScoreRatio { get; set; }
 
+        /// <summary>
+        /// The bonus portion of the legacy (ScoreV1) total score.
+        /// </summary>
+        public int LegacyBonusScore { get; set; }
```

Comment:
I don't know the implications that come with this (is it required that we write these out to the database?). Would want @smoogipoo to chime in here.

Reply:
That depends on the implementation. If it's required for score conversion, then yes it will need to be written to the database.

Doing so is simple, by just changing the `ToDatabaseAttributes` and `FromDatabaseAttributes` methods, though whether you want to add 2 new properties along with the 3 new ones added to facilitate the current sv2 conversion, is up to your discretion. And whether the current ones are still required by the new conversion in this PR.  

As a general rule, I've tried to keep these attributes as lean as possible, purely for database size concerns, and I would suggest the same to be done here.

Which also brings me to - is `LegacyMaxCombo` required in the first place. The difficulty calculators are pretty much legacy-only right now, so doesn't this duplicate `MaxCombo`, and if so - why not? Because if not then it's an issue.

Reply:
#### For the LegacyBonusScore:
The problem when converting legacy scores is we don't know how much of the score comes from the bonus and how much comes from something else. Thus there is three approaches:
- Assume the minimum possible bonus
- Assume the maximum possible bonus
- Assume the same ratio of bonus as in the theoretically best possible score on the map.

The previous algo (by smoogi?) was assuming the 1st option (minimum bonus) everywhere. I went with the third approach for the ComboScore, and the 1st for the BonusScore to reduce the chance of overestimating scores with low combo and high spinner bonus but still keeping the conversion close in most cases. Alternatively, the worst case scenario can be computed using the 2nd approach for ComboScore and 1st for BonusScore, but the problem is that then you could end up with a ComboScore of 0 after conversion in some extreme scenarios, even though you had comboed a bit.

#### For LegacyMaxCombo:
lazer's MaxCombo is not the same as the Legacy one (unless we match stable and give combo on slider-ends ? Idk if there's any other difference though) but I have no idea if the "MaxCombo" in DifficultyAttributes is supposed to be the legacy one or the new one though.

Reply:
> but I have no idea if the "MaxCombo" in DifficultyAttributes is supposed to be the legacy one or the new one though.

Have you tried it? When it's relevant (i.e. storing to the database), diffcalc attaches the classic mod, which sets this to false:

https://github.com/ppy/osu/blob/master/osu.Game.Rulesets.Osu/Objects/Slider.cs#L270

... which gives sliders a combo-affecting judgement (effectively the slider tail).

Reply:
> Have you tried it?
Nope. What I meant is that I expected MaxCombo to be the lazer one (without classic mod) when I wrote this PR (especially since other legacy fields have "Legacy" in the name), but I might have been completely wrong indeed.

Reply:
As for `LegacyBonusScore`, I take it that the "ratio" version that I have is still required for the other rulesets?

Reply:
I don't remember but either that or I wasn't sure if I would use it at first and then forgot about it. It might be simpler to just have a LegacyBonusScore and NewBonusScore (or whatever you name it) though if the ratio is needed, but I wanted to minimize the changes.

Reply:
Since it's recently come to my(/our?) attention that the size of this table in production is 117GB, this is definitely something to consider.

Reply:
Are these legacy attributes required to be part of this PR?